server/src/main/java/org/opensearch/index/remote/RemoteStoreEnums.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.remote;
+
+import org.opensearch.common.annotation.PublicApi;
+import org.opensearch.common.blobstore.BlobPath;
+import org.opensearch.common.hash.FNV1a;
+import org.opensearch.index.remote.RemoteStorePathStrategy.PathInput;
+
+import java.util.Locale;
+import java.util.Set;
+
+import static org.opensearch.index.remote.RemoteStoreEnums.DataType.DATA;
+import static org.opensearch.index.remote.RemoteStoreEnums.DataType.METADATA;
+
+/**
+ * This class contains the different enums related to remote store like data categories and types, path types
+ * and hashing algorithm.
+ *
+ * @opensearch.api
+ */
+public class RemoteStoreEnums {
+
+    /**
+     * Categories of the data in Remote store.
+     */
+    @PublicApi(since = "2.14.0")
+    public enum DataCategory {
+        SEGMENTS("segments", Set.of(DataType.values())),
+        TRANSLOG("translog", Set.of(DATA, METADATA));
+
+        private final String name;
+        private final Set<DataType> supportedDataTypes;
+
+        DataCategory(String name, Set<DataType> supportedDataTypes) {
+            this.name = name;
+            this.supportedDataTypes = supportedDataTypes;
+        }
+
+        public boolean isSupportedDataType(DataType dataType) {
+            return supportedDataTypes.contains(dataType);
+        }
+
+        public String getName() {
+            return name;
+        }
+    }
+
+    /**
+     * Types of data in remote store.
+     */
+    @PublicApi(since = "2.14.0")
+    public enum DataType {
+        DATA("data"),
+        METADATA("metadata"),
+        LOCK_FILES("lock_files");
+
+        private final String name;
+
+        DataType(String name) {
+            this.name = name;
+        }
+
+        public String getName() {
+            return name;
+        }
+    }
+
+    /**
+     * Enumerates the types of remote store paths resolution techniques supported by OpenSearch.
+     * For more information, see <a href="https://github.com/opensearch-project/OpenSearch/issues/12567">Github issue #12567</a>.
+     */
+    @PublicApi(since = "2.14.0")
+    public enum PathType {
```

Comment:
should it contain `PathHashAlgorithm hashAlgorithm` as a member as well. It shouldn't take `hashAlgorithm` everytime it generates path and should only take `pathInput` as an input .

Reply:
We want to have the door for extensibility on being able to add more hash algorithms in future. PathInput only contains details about the characteristics about the path (& it's data). Hash algo is not part of that. To the caller, they only make the calls to RemoteStorePathStrategy class.